pkg/pillar/hypervisor/kvm.go

```diff
@@ -1343,31 +1352,54 @@ func (ctx KvmContext) VirtualTPMTerminate(domainName string) error {
 func (ctx KvmContext) VirtualTPMTeardown(domainName string) error {
 	domainUUID, _, _, err := types.DomainnameToUUID(domainName)
 	if err != nil {
-		return fmt.Errorf("failed to extract UUID from domain name (vTPM teardown): %v", err)
+		return fmt.Errorf("failed to extract UUID from domain name: %v", err)
 	}
-	if err := requestVtpmPurge(domainUUID); err != nil {
+	if err := requestvTPMPurge(domainUUID); err != nil {
 		return fmt.Errorf("failed to purge vTPM for domain %s: %w", domainName, err)
 	}
 
 	return nil
 }
 
-func requestVtpmLaunch(id uuid.UUID, wk *utils.WatchdogKick, timeoutSeconds uint) error {
-	conn, err := net.Dial("unix", types.VtpmdCtrlSocket)
+func vtpmClientUDSTransport() *http.Transport {
+	return &http.Transport{
+		DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+			return net.Dial("unix", types.VtpmdCtrlSocket)
+		},
+	}
+}
+
+func makeRequest(client *http.Client, endpoint, id string) (string, error) {
+	url := fmt.Sprintf("http://unix/%s?id=%s", endpoint, id)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
-		return fmt.Errorf("failed to connect to vTPM control socket: %w", err)
+		return "", fmt.Errorf("error when creating request: %v", err)
 	}
-	defer conn.Close()
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("request failed: %v", err)
+	}
+	defer resp.Body.Close()
 
-	pidPath := fmt.Sprintf(types.SwtpmPidPath, id.String())
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("error when reading response body: %v", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return string(body), fmt.Errorf("received status code %d", resp.StatusCode)
```

Comment:
How about adding the url to this and the previous two errror strings?

Reply:
added.